Realm/RLMRealmConfiguration.mm

```diff
@@ -233,6 +233,8 @@ - (void)setDeleteRealmIfMigrationNeeded:(BOOL)deleteRealmIfMigrationNeeded {
     if (deleteRealmIfMigrationNeeded) {
         if (self.readOnly) {
             @throw RLMException(@"Cannot set `deleteRealmIfMigrationNeeded` when `readOnly` is set.");
+        } else if (isSync(_config) && !_config.sync_config->partition_value.empty()) {
```

Comment:
Is there any way Sync can be enabled with an empty partition value?

Reply:
Also won't this need some `#if REALM_ENABLE_SYNC` guards here?

Reply:
Fixed, both